### PR TITLE
Add instructions on how to use the installed libraries.

### LINKED ISSUE #2086: README+INSTALL should describe how to use the libraries once compiled + installed.
This was feedback from a live customers. They found the instructions to install the libraries, but we do not have anything in the README.md or the INSTALL.md files on how to use the library in their own projects once the library is installed.

We need something like this but with details for Bazel, and maybe CMake super projects too:

https://stackoverflow.com/questions/54782935/how-do-i-compile-c-code-including-google-cloud-storage-library

### PR DESCRIPTION
**Pull request by coryan**
This fixes #2086.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2335)

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -24,7 +24,96 @@ Unfortunately getting your system to this state may require multiple steps,
...
+Once you have installed `google-cloud-cpp` you can use the libraries in your
+own Make-based projects using `pkg-config`:
+
+```Makefile
+GCS_CXXFLAGS   := $(shell pkg-config storage_client --cflags)
+GCS_CXXLDFLAGS := $(shell pkg-config storage_client --libs-only-L)
+GCS_LIBS := $(shell pkg-config storage_client --libs-only-l) -lcrc32c
+
+my_storage_program: my_storage_program.cc
+	$(CXXLD) $(CXXFLAGS) $(GCS_CXXFLAGS) $(GCS_CXXLDFLAGS) -o $@ $^ $(GCS_LIBS)
```
Comment (devbww) [inline INSTALL.md]: s/CXXLD/CXX/?

[All this in the generate script I suppose.]

File: INSTALL.md [hunk 2]
```diff
@@ -24,7 +24,96 @@ Unfortunately getting your system to this state may require multiple steps,
...
+
+CBT_CXXFLAGS   := $(shell pkg-config bigtable_client --cflags)
+CBT_CXXLDFLAGS := $(shell pkg-config bigtable_client --libs-only-L)
+CBT_LIBS := $(shell pkg-config bigtable_client --libs-only-l) -lcrc32c
+
+my_storage_program: my_storage_program.cc
```
Comment (devbww) [inline INSTALL.md]: my_bigtable_program?

File: INSTALL.md [hunk 3]
```diff
@@ -24,7 +24,96 @@ Unfortunately getting your system to this state may require multiple steps,
...
+	$(CXXLD) $(CXXFLAGS) $(CBT_CXXFLAGS) $(CBT_CXXLDFLAGS) -o $@ $^ $(CBT_LIBS)
```
Comment (devbww) [inline INSTALL.md]: s/CXXLD/CXX/?

File: google/cloud/bigtable/doc/bigtable-main.dox [hunk 4]
```diff
@@ -93,26 +93,36 @@ cc_binary(
...
 
-@note This assumes that you have downloaded and installed the dependencies of
-  the library. Please check the top-level README on [GitHub][github-link] for
-  details.
-
-Compile and install the library using the normal cycle:
+Follow the instructions on the top-level INSTALL.md file on
+[GitHub][github-link] for details on how to compile and install the library for
+your platform and distribution. Once the library is installed, you can use it in
+your `CMakeLists.txt` file as any other package:
```
Comment (devbww) [inline google/cloud/bigtable/doc/bigtable-main.dox]: s/as/like/?

File: google/cloud/bigtable/doc/bigtable-main.dox [hunk 5]
```diff
@@ -93,26 +93,36 @@ cc_binary(
...
 
-Now that the library is installed, you can use it in your CMake list files:
+#### Integrating with Make
 
-```
-find_package(bigtable_client 0.2.0 REQUIRED)
+The installation instructions on the top-level INSTALL.md file on
+[GitHub][github-link] also create `pkg-config` support files for application
+developers that prefer to use `Make` as their build system. Once the library
+is installed, you can use it in your `Makefile` as any other `pkg-config`
```
Comment (devbww) [inline google/cloud/bigtable/doc/bigtable-main.dox]: s/as/like/?

File: google/cloud/bigtable/doc/bigtable-main.dox [hunk 6]
```diff
@@ -93,26 +93,36 @@ cc_binary(
...
-target_link_libraries(my_program bigtable_client)
+```
+CBT_DEPS := bigtable_client
+CBT_CXXFLAGS   := $(shell pkg-config $(CBT_DEPS) --cflags)
+CBT_CXXLDFLAGS := $(shell pkg-config $(CBT_DEPS) --libs-only-L)
+CBT_LIBS       := $(shell pkg-config $(CBT_DEPS) --libs-only-l)
+
+# A target using the Cloud Bigtable C++ client library.
+bigtable_install_test: bigtable_install_test.cc
+	$(CXXLD) $(CXXFLAGS) $(CBT_CXXFLAGS) $(GCS_CXXLDFLAGS) -o $@ $^ $(CBT_LIBS)
```
Comment (devbww) [inline google/cloud/bigtable/doc/bigtable-main.dox]: s/CXXLD/CXX/?

File: google/cloud/storage/doc/storage-main.dox [hunk 7]
```diff
@@ -175,28 +175,38 @@ cc_binary(
...
 
-@note This assumes that you have downloaded and installed the dependencies of
-  the library. Please check the top-level README on [GitHub][github-link] for
-  details.
-
-Compile and install the library using the normal cycle:
+Follow the instructions on the top-level INSTALL.md file on
+[GitHub][github-link] for details on how to compile and install the library for
+your platform and distribution. Once the library is installed, you can use it in
+your `CMakeLists.txt` file as any other package:
```
Comment (devbww) [inline google/cloud/storage/doc/storage-main.dox]: s/as/like/?

File: google/cloud/storage/doc/storage-main.dox [hunk 8]
```diff
@@ -175,28 +175,38 @@ cc_binary(
...
+#### Integrating with Make
 
-```
-find_package(Threads REQUIRED)
-find_package(CURL REQUIRED)
-find_package(storage_client 0.4.0 REQUIRED)
+The installation instructions on the top-level INSTALL.md file on
+[GitHub][github-link] also create `pkg-config` support files for application
+developers that prefer to use `Make` as their build system. Once the library
+is installed, you can use it in your `Makefile` as any other `pkg-config`
```
Comment (devbww) [inline google/cloud/storage/doc/storage-main.dox]: s/as/like/?

File: google/cloud/storage/doc/storage-main.dox [hunk 9]
```diff
@@ -175,28 +175,38 @@ cc_binary(
...
+# Configuration variables to compile and link against the Google Cloud Storage
+# C++ client library.
+GCS_DEPS := storage_client
+GCS_CXXFLAGS   := $(shell pkg-config $(GCS_DEPS) --cflags)
+GCS_CXXLDFLAGS := $(shell pkg-config $(GCS_DEPS) --libs-only-L)
+GCS_LIBS       := $(shell pkg-config $(GCS_DEPS) --libs-only-l)
+
+# A target using the Google Cloud Storage C++ client library.
+storage_install_test: storage_install_test.cc
+	$(CXXLD) $(CXXFLAGS) $(GCS_CXXFLAGS) $(GCS_CXXLDFLAGS) -o $@ $^ $(GCS_LIBS)
```
Comment (devbww) [inline google/cloud/storage/doc/storage-main.dox]: s/CXXLD/CXX/?

File: INSTALL.md (hunk 1 above)
Comment (coryan) [reply INSTALL.md]: Fixed.

File: INSTALL.md (hunk 2 above)
Comment (coryan) [reply INSTALL.md]: Fixed.

File: INSTALL.md (hunk 3 above)
Comment (coryan) [reply INSTALL.md]: Fixed.

File: google/cloud/bigtable/doc/bigtable-main.dox (hunk 4 above)
Comment (coryan) [reply google/cloud/bigtable/doc/bigtable-main.dox]: Fixed.

File: google/cloud/bigtable/doc/bigtable-main.dox (hunk 5 above)
Comment (coryan) [reply google/cloud/bigtable/doc/bigtable-main.dox]: Fixed.

File: google/cloud/bigtable/doc/bigtable-main.dox (hunk 6 above)
Comment (coryan) [reply google/cloud/bigtable/doc/bigtable-main.dox]: Fixed.

File: google/cloud/storage/doc/storage-main.dox (hunk 7 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-main.dox]: Fixed.

File: google/cloud/storage/doc/storage-main.dox (hunk 8 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-main.dox]: Fixed

File: google/cloud/storage/doc/storage-main.dox (hunk 9 above)
Comment (coryan) [reply google/cloud/storage/doc/storage-main.dox]: Fixed.

**Pull request merged**